src/main/java/com/gdschongik/gdsc/domain/membership/domain/Membership.java

```diff
@@ -83,11 +84,20 @@ private static void validateMembershipApplicable(Member member) {
 
     public void verifyPaymentStatus() {
         this.regularRequirement.updatePaymentStatus(VERIFIED);
+
+        registerEvent(new MemberAdvanceToRegularEvent(member.getDiscordUsername(), member.getId()));
     }
 
     // 데이터 전달 로직
 
-    public boolean isAdvanceToRegularAvailable() {
+    public boolean isAdvanceRequirementAllSatisfied() {
         return this.regularRequirement.isAllVerified();
     }
+
+    public void validateAdvanceRequirement() {
+        if (!isAdvanceRequirementAllSatisfied()) {
+            return;
+        }
+        throw new CustomException(MEMBERSHIP_ALREADY_VERIFIED);
+    }
```

Comment:
```suggestion
    public void validateAdvanceRequirement() {
        if (isAdvanceRequirementAllSatisfied()) {
            throw new CustomException(MEMBERSHIP_ALREADY_VERIFIED);
        }
    }
```

이런식으로 줄이면 더 알아보기 좋을 것 같습니다

Reply:
코드들이 저런 형식으루 많이 되어있어서 맞춰서 했습니답!

Reply:
슬기님이 제안해주신 버전이 더 좋아보입니다.
이런 식으로 되어있지 않은 패턴이 있나요?
만약 그런 케이스가 있으면 리팩토링하거나 TODO 주석 남겨주시면 좋을듯 합니다

Reply:
validate 검증 함수에 몇개 되어있는 것 같은데 주석 처리 남기고 커밋하겠습니다